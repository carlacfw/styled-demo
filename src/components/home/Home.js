import React from 'react'
import styled from 'styled-components'

const P = styled.p`
  font-family: 'Courier New', Courier, monospace;
  font-size: 0.9em;
`
const HomeWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 30px 0;
  width: 60%;
`

const Home = () => {
  return (
    <HomeWrapper>
      <P>
        This is a demo app designed to demonstrate a few of the fantastic
        features of{' '}
        <a
          href="https://www.styled-components.com/"
          target="_blank"
          rel="noopener noreferrer"
        >
          Styled Components
        </a>{' '}
        library to WellingtonJS. The goal is to demonstrate the power of writing
        CSS in JS to solve comon web development problems such as building
        froms, and using variabled and composition to asist in code reuse.
      </P>
      <P>
        What this is not is a design or CSS masterclass. This was produced
        quickly to demonstrate library features, production code will need
        thought out design and more refined styling.
      </P>
      <P>
        I hope that this encourages to try using CSS in JS using librarys like
        styled components.
      </P>
    </HomeWrapper>
  )
}

export default Home
