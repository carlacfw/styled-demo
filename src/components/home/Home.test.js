import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'
describe('The Home page ', () => {
  it('It renders the home page correctly', () => {
    const wrapper = shallow(<Home />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
