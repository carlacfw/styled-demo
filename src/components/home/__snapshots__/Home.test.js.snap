// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home page  It renders the home page correctly 1`] = `
<styled.div>
  <styled.p>
    This is a demo app designed to demonstrate a few of the fantastic features of
     
    <a
      href="https://www.styled-components.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Styled Components
    </a>
     
    library to WellingtonJS. The goal is to demonstrate the power of writing CSS in JS to solve comon web development problems such as building froms, and using variabled and composition to asist in code reuse.
  </styled.p>
  <styled.p>
    What this is not is a design or CSS masterclass. This was produced quickly to demonstrate library features, production code will need thought out design and more refined styling.
  </styled.p>
  <styled.p>
    I hope that this encourages to try using CSS in JS using librarys like styled components.
  </styled.p>
</styled.div>
`;
